src/awkward/forms/form.py

```diff
@@ -65,12 +67,22 @@ def from_dict(input: dict) -> Form:
         )
 
     elif input["class"] == "RecordArray":
-        if isinstance(input["contents"], dict):
+        # New serialisation
+        if "fields" in input:
+            if isinstance(input["contents"], Mapping):
```

Comment:
Note: since this is coming from `json.loads` or equivalent, we can make a hard assumption that JSON objects are dicts. Being more general doesn't hurt, but it's why the rest of this function is checking `isinstance(_, list)` and `isinstance(_, dict)` everywhere.

Reply:
This warning is different to the rest of the function body, which (as you point out) is much stricter with types.

In reality, this is a high level functions that users can call. My temptation is to relax the constraints from `dict` to `Mapping`, and `list` to `sized_iterable`. How do you feel about that?